Libraries/CommonFunctions.psm1

```diff
@@ -117,19 +70,36 @@ Function Select-TestCases($TestXMLs, $TestCategory, $TestArea, $TestNames, $Test
     $WildCards = @('^','.','[',']','?','+','*')
     $ExcludedTestsCount = 0
 
-    # if the expected filter parameter is 'ALL' or empty ("ALL" -imatch "" return $True), the actually filter used in this function will be '*', except for '$ExcludedTests'
-    if ("All" -imatch $TestCategory)   { $TestCategory = "*" }
-    if ("All" -imatch $TestArea)       { $TestArea = "*" }
-    if ("All" -imatch $TestNames)      { $TestNames = "*" }
-    if ("All" -imatch  $TestTag)       { $TestTag = "*" }
-    if ("All" -imatch  $TestPriority)  { $TestPriority = "*" }
-    if ("All" -imatch  $TestSetup)     { $TestSetup = "*" }
+    # if the expected filter parameter is 'All' (case insensitive) or empty or $null, the actually filter used in this function will be '*', except for '$ExcludedTests'
+    if (!$TestCategory -or ($TestCategory -eq "All"))   { $TestCategory = "*" }
```

Comment:
Can we add else here to update all arrays ?

Reply:
updated